apps/dashboard/test/apps/nav_bar_test.rb

```diff
@@ -0,0 +1,142 @@
+require 'test_helper'
+
+class NavBarTest < ActiveSupport::TestCase
+
+  def setup
+    # Mock the sys installed applications
+    OodAppkit.stubs(:clusters).returns(OodCore::Clusters.load_file('test/fixtures/config/clusters.d'))
+    SysRouter.stubs(:base_path).returns(Rails.root.join("test/fixtures/sys_with_gateway_apps"))
+  end
+
+  test "NavBar.items should return navigation template when nav_item is a string matching a static link" do
+    nav_item = "all_apps"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/all_apps",  result[0].partial_path
+  end
+
+  test "NavBar.items should return navigation link when nav_item is a string matching one app token" do
+    nav_item = "sys/bc_jupyter"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/link",  result[0].partial_path
+    assert_equal "Jupyter Notebook",  result[0].title
+    assert_equal "/batch_connect/sys/bc_jupyter/session_contexts/new",  result[0].url
+  end
+
+  test "NavBar.items should return navigation group when nav_item is a string matching multiple app tokens" do
+    nav_item = "sys/bc_*"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/group",  result[0].partial_path
+    assert_equal "Apps",  result[0].title
+    assert_equal 4,  result[0].apps.size
+  end
```

Comment:
This behavior kinda caught me by surprise, but I see what's going on now. We're decorating (possibly) apps that have already been decorated. When re refactor `FeaturedApps` this may change, though it's unclear to me how.

Reply:
Yes, this is a bit tricky.
When `nav_item` is a String, I think we should not allow it if it matches more than one application. We will not be able to determine how to render this use case. It will only work if all the matched apps have the same category.

Currently only one group comes back as Featured Apps is overriding  category to "Apps " and subcategory to "Pinned Apps".